contracts/Registrar.sol

```diff
@@ -97,67 +156,61 @@ contract Registrar {
+
+    /// Set the owner of the domain.
+    function setDomainOwner(address newOwner) public adminOnly {
+        IERC721 ethRegistrar = IERC721(ens.owner(ethNode));
 
-    // ADMIN FUNCTIONS
+        ens.setOwner(radNode, newOwner);
+        ethRegistrar.transferFrom(address(this), newOwner, tokenId);
+        commitments.setOwner(newOwner);
 
-    /// Set the USD registration fee.
-    function setUsdRegistrationFee(uint256 fee) public adminOnly {
-        registrationFeeUsd = fee;
+        emit DomainOwnershipChanged(newOwner);
     }
 
-    /// Set the radicle registration fee.
-    function setRadRegistrationFee(uint256 fee) public adminOnly {
-        registrationFeeRad = fee;
+    /// Set a new resolver for radicle.eth.
+    function setDomainResolver(address resolver) public adminOnly {
+        ens.setResolver(radNode, resolver);
+        emit ResolverChanged(resolver);
     }
 
-    /// Set the price oracle.
-    function setPriceOracle(address _oracleAddress) public adminOnly {
-        require(
-            oracleAddress != address(0),
-            "Registrar::setPriceOracle: oracle address cannot be zero"
-        );
-        oracleAddress = _oracleAddress;
+    /// Set a new ttl for radicle.eth
+    function setDomainTTL(uint64 ttl) public adminOnly {
+        ens.setTTL(radNode, ttl);
+        emit TTLChanged(ttl);
     }
 
-    /// Set the exchange.
-    function setExchange(address _exchangeAddress) public adminOnly {
-        require(
-            exchangeAddress != address(0),
-            "Registrar::setExchange: exchange address cannot be zero"
-        );
-        exchangeAddress = _exchangeAddress;
+    /// Set the minimum commitment age
+    function setMinCommitmentAge(uint256 amt) public adminOnly {
+        minCommitmentAge = amt;
+        emit MinCommitmentAgeChanged(amt);
     }
 
-    /// Set the owner of the domain.
-    function setDomainOwner(address newOwner) public adminOnly {
-        // The name hash of 'eth'
-        bytes32 ethNode = 0x93cdeb708b7545dc668eb9280176169d1c33cfd8ed6f04690a0bcc88a93fc4ae;
-        address ethRegistrarAddr = ens.owner(ethNode);
-        require(
-            ethRegistrarAddr != address(0),
-            "Registrar::setDomainOwner: no registrar found on ENS for the 'eth' domain"
-        );
-        ens.setRecord(domain, newOwner, newOwner, 0);
-        IERC721 ethRegistrar = IERC721(ethRegistrarAddr);
-        ethRegistrar.transferFrom(address(this), newOwner, tokenId);
+    /// Set a new registration fee
+    function setRegistrationFee(uint256 amt) public adminOnly {
+        fee = amt;
+        emit RegistrationFeeChanged(amt);
     }
 
     /// Set a new admin
-    function setAdmin(address _admin) public adminOnly {
-        admin = _admin;
+    function setAdmin(address newAdmin) public adminOnly {
+        admin = newAdmin;
+        emit AdminChanged(newAdmin);
+    }
+
+    /// Set the address for the ENS registry
+    function setEns(address newEns) public adminOnly {
```

Comment:
Can we realistically expect this to be ever used? If the ENS is moved, it's probably due to some API changes, which will require a full registrar update anyway. If we remove this function, we'll save some gas on accessing the `ens` variable.

Reply:
I'm inclined to agree. If another migration were to happen it could be dealt with by transferring ownership to a new Registrar.